onnx/onnx-data.in.proto

```diff
@@ -99,3 +99,40 @@ message MapProto {
   // TENSOR, SPARSE_TENSOR, MAP, SEQUENCE.
   optional SequenceProto values = 5;
 }
+
+// Optional
+//
+//
+message OptionalProto {
+
+  optional string name = 1;
+
+  enum DataType {
+    UNDEFINED = 0;
+    TENSOR = 1;
+    SPARSE_TENSOR = 2;
+    SEQUENCE = 3;
+    MAP = 4;
```

Comment:
(Comment meant mainly for SequenceProto, but also here for completeness) We should add "OPTIONAL = 5" to this enum, and add a field "repeated OptionalProto optional_values" (for SequenceProto) and, similarly, "optional OptionalProto optional_value" here ... even though an optional value of optional type is unlikely to be used, it would help to be complete.